source/set-required-deep.d.ts

```diff
@@ -0,0 +1,47 @@
+import type {OmitDeep} from './omit-deep';
+import type {Paths} from './paths';
+import type {PickDeep} from './pick-deep';
+import type {RequiredDeep} from './required-deep';
+import type {SimplifyDeep} from './simplify-deep';
+
+/**
+Create a type that makes the given keys required. You can specifiy deeply nested key paths. The remaining keys are kept as is.
+
+Use-case: You want to define as required only a some nested keys inside a model.
+
+@example
+```
+import type {SetRequiredDeep} from 'type-fest';
+
+type Foo = {
+	a?: number;
+	b?: string;
```

Comment:
add a array example is better.
```suggestion
     	b?: string;
	array?: {
	  a?: number;
	}[];
```